simple_use_case/nextflow/simplecase.nf

```diff
@@ -15,52 +18,70 @@ process generateMesh {
     conda "../source/envs/preprocessing.yaml"
 
     input:
-    file geo from geo_ch
+    file geofile
 
     output:
-    file "unit_square.msh" into msh_ch
+    file "unit_square.msh" into mesh_msh
 
     """
-    gmsh -2 -order 1 -format msh2 $geo -o unit_square.msh
+    gmsh -2 -setnumber domain_size ${params.domainSize} $geofile -o unit_square.msh
     """
 }
 
 
 process convertToXDMF {
-    // convert any msh file to xdmf
+    // convert .msh file to .xdmf
 
     conda "../source/envs/preprocessing.yaml"
 
     input:
-    file msh from msh_ch
+    file mesh from mesh_msh
 
     output:
-    file "unit_square.h5" into h5_ch
-    file "unit_square.xdmf" into xdmf_ch
+    file "unit_square.h5" into mesh_h5
+    file "unit_square.xdmf" into mesh_xdmf
 
     """
-    meshio convert $msh unit_square.xdmf
+    meshio convert $mesh unit_square.xdmf
     """
 }
 
 
 process solvePoisson {
-    // solve poisson equation using fenics code
+    // solve poisson equation using fenics
 
     conda "../source/envs/processing.yaml"
 
     // multiple inputs from different channels
     input:
-    file fenics_code from fenics_ch
-    file meshdata from h5_ch
-    file mesh from xdmf_ch
+    file fenics_code
+    file meshdata from mesh_h5
+    file mesh from mesh_xdmf
 
     output:
-    file "poisson.pvd" into pvd_ch
-    file "poisson*.vtu" into vtu_ch
+    file "poisson.pvd" into poisson_pvd
+    file "poisson*.vtu" into poisson_vtu
+    stdout poisson_stdout
 
     """
-    python $fenics_code --mesh $mesh --degree ${params.degree} --outputfile poisson.pvd
+    python $fenics_code --mesh $mesh --degree 2 --outputfile poisson.pvd
+    """
+}
+
+
+process readNumberOfDofs {
+    input:
+    val x from poisson_stdout
+
+    output:
+    stdout number_of_dofs
+
+    """
+    #!/usr/bin/python3
+    s = '''${x.replaceAll("\\n", "&")}'''
+    dof_string = s.split("Number of dofs used:")[1]
+    num_dofs = dof_string.split("&")[0]
+    print(int(num_dofs))
```

Comment:
So this scrip snippet prints the number of dofs, which is returned as `stdout`, I guess it is not possible to return an integer directly?  Nextflow also works file based, right?

Reply:
It's all about [Channels](https://www.nextflow.io/docs/latest/channel.html#channel-types). These communicate the flow of data between processes. You can have input/output of type value (this could either be a string or an integer or a list of string/integer etc) or file etc. from a certain channel. See this example for [output values](https://www.nextflow.io/docs/latest/process.html#output-values).
At the time of writing using `stdout` seemed necessary since I need to parse the string first before passing on the value of number of dofs to some output channel.

Reply:
in the example [output values](https://www.nextflow.io/docs/latest/process.html#output-values) I probably could do the parsing using bash and then `echo $x > file` as the last line of the execution block. This way I could use `val x into receiver` as in the example, but I think essentially a *file* would be created.

Reply:
yeah, nevermind. Ultimately it is treated as string anyway when the value is passed to the macro preparation script :)